main/src/ca/uwaterloo/flix/language/phase/Desugar.scala

```diff
@@ -1460,6 +1465,42 @@ object Desugar {
     DesugaredAst.Expr.Lambda(fparam, body, loc0.asSynthetic)
   }
 
+  /**
+    * Desugars [[WeededAst.Expr.IfThen]] to [[Expr.IfThenElse]]
+    * {{{
+    * if (exp1) {
+    *   exp2
+    * }
+    * }}}
+    * desugars to
+    * {{{
+    * if (exp1) {
+    *   exp2;
+    *   ()
+    * } else {
+    *  ()
+    * }
+    * }}}
+    */
+  private def desugarIfThen(guard: DesugaredAst.Expr, then0: DesugaredAst.Expr, loc: SourceLocation): DesugaredAst.Expr = {
```

Comment:
We're trying to move away from desugaring. We should consider instead making the then branch an Option and then propagating the change through all the ASTs.